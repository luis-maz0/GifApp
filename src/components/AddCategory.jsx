import { useState } from "react"

export const AddCategory = ()=>{
    const [inputValue, setInputValue] = useState(); 
    const onInputValue = ({ target })=>{
        console.log( target.value); 
        setInputValue( target.value );
    }
    const onSubmit = (e)=>{
        e.preventDefault(); 
        console.log( inputValue)
    }
    return(
        <form onSubmit={ onSubmit }>
            <input
                type="text"
                placeholder="Search gif"
                onChange={ onInputValue }
            />
        </form>
    )
}