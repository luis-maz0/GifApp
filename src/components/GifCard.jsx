export const GifCard = ({ category }) => {
    const amountGifs = [1,2,3,4,5]; //Traer los gifs con Fetch. 
    return (
    <>
        <h3>{ category }</h3>
        { 
            amountGifs.map( gif => <p key={ gif }>{ gif }</p> )
        }
    </>)
}
