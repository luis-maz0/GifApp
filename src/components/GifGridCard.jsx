import PropTypes from "prop-types"; 

export const GifGridCard = ({ title, url }) => {
  return (
    <div className="card">
        <img src={url} alt={title} />
    </div>
  )
}
//PropTypes
GifGridCard.propTypes = {
  title: PropTypes.string.isRequired, 
  url: PropTypes.string.isRequired,
}
