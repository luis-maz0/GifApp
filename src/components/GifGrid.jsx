import { useEffect, useState } from "react";

import { GifGridCard } from "./GifGridCard";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category }) => {
    
    const { gifs, isLoading} = useFetchGifs( category )

    return (
        <>
            <h3>{category}</h3>
            <div className="card-grid">
                {
                    gifs.map(gif =>
                        <GifGridCard
                            key={gif.id}
                            {...gif}
                        />
                    )}
            </div>
        </>)
}
