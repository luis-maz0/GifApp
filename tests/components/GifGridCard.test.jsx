import { GifGridCard } from "../../src/components/GifGridCard"
import { render, screen} from "@testing-library/react"

describe('Pruebas en <GifGridCar/>', () => { 
    
    const title = "One Piece";
    const url = "https://onepiece.com/luffy.jpg";
    
    test('Match con el snapshot', () => {
        const {container } = render( <GifGridCard title={title} url={url}/> )
        expect( container ).toMatchSnapshot()
    })
    
    test('Debe mostrar la imm con la URL y ALT indicado', () => {
        render( <GifGridCard title={title} url={url}/>)
        screen.debug()
        expect( screen.getByRole("img").src).toBe(url)
        expect( screen.getByRole("img").alt).toBe(title)
      })
 })