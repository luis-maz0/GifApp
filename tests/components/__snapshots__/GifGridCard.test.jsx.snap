// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridCar/> Match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Piece"
      src="https://onepiece.com/luffy.jpg"
    />
  </div>
</div>
`;
